polaris-react/src/components/IndexTable/IndexTable.module.scss

```diff
@@ -159,6 +159,10 @@ $loading-panel-height: 53px;
     cursor: auto;
   }
 
+  &.TableRow-borderless:not(:first-child) {
```

Comment:
I'd refer to Polaris team as the experts on this. 

My initial thought here is if we are making the divider configurable thus moving the responsibility to the consumer - then we should probably not force it on certain elements and let the consumer take control of it entirely.

Reply:
I think that's make sense, thank you for the suggestion!